parlai/mturk/tasks/acute_eval/run.py

```diff
@@ -112,341 +102,459 @@ def add_args(from_argv=False):
+        """
+        onboarding_todo = list(range(len(self.onboarding_tasks)))
+        random.shuffle(onboarding_todo)
+        self.worker_data[worker_id] = self.worker_data.get(
+            worker_id,
+            {
+                'tasks_completed': [],
+                'conversations_seen': [],
+                'onboarding_todo': onboarding_todo,
+                'hits_completed': 0,
+            },
+        )
+        return self.worker_data[worker_id]
+
+    def _supplement_opt(self):
+        """
+        Add additional args to opt.
+
+        Useful to add relevant options after args are parsed.
+        """
+        self.opt.update(
+            {
+                'task': os.path.basename(os.path.dirname(os.path.abspath(__file__))),
+                'task_description': {
+                    'num_subtasks': self.opt['subtasks_per_hit'],
+                    'question': self.opt['question'],
+                },
+                'frontend_version': 1,
+            }
+        )
+        self.opt.update(self.opt['task_config'])
+
+    def set_block_qual(self, task_group_id: str):
+        """
+        Set block qualification if necessary.
+
+        :param task_group_id:
+            task id used to set block qualification, if necessary.
+        """
+        if (
+            self.opt['block_on_onboarding_fail']
+            and self.opt['block_qualification'] is None
+        ):
+            self.opt['block_qualification'] = task_group_id
             warn_once(
                 "No block_qualification set in opt, automatically creating "
                 "new qualification {}".format(task_group_id)
             )
 
-        # Set up the sockets and threads to receive workers
-        mturk_manager.ready_to_accept_workers()
+    def _load_conversation_data(self):
+        """
+        Load conversation data.
+
+        Loads in the data from the pairs filepath.
+        """
+        pairs_path = self.opt.get('pairings_filepath')
+        if not os.path.exists(pairs_path):
+            raise RuntimeError('You MUST specify a valid pairings filepath')
+
+        with open(pairs_path) as pf:
+            for i, l in enumerate(pf.readlines()):
+                convo_pair = json.loads(l.strip())
+                eval_speakers = [
+                    s
+                    for d in convo_pair['dialogue_dicts']
+                    for s in d['speakers']
+                    if s in convo_pair['speakers_to_eval']
+                ]
+                # make sure order is preserved
+                assert eval_speakers == convo_pair['speakers_to_eval']
+                task = {
+                    'task_specs': {
+                        'conversation_order': random.choice([[0, 1], [1, 0]]),
+                        's1_choice': self.opt['s1_choice'],
+                        's2_choice': self.opt['s2_choice'],
+                        'question': self.opt['question'],
+                        # TODO: make sure this doesn't break javascript
```

Comment:
@klshuster was this my comment or yours?

Reply:
mine, and it does not break javascript so we're good 😄